elastic/fleet-server
# Fix the index monitor checkpoint tracking for when the documents are deleted

## What is the problem this PR solves?

This fixes the bug with the fleet server actions monitor that doesn't account correctly on the document sequence number increment when the expired action is deleted. This leads to the monitor checkpoint getting out of sync (lagging behind) with the index checkpoint, and causes the monitor to go into the tight loop of awaiting for checkpoint advance (which returns immediately and fetching the actions documents.

"ESS deployment being flooded by requests to /.fleet-actions/_fleet/_fleet_search. 1.3M requests over last 2 hours when customer workload is mainly Logstash indexing."

The current workaround for this would be either restart the fleet-server or push the new action, like change the logging level for example.

## How does this PR solve the problem?

Update monitor checkpoint value from the "wait advance" response upon the last fetch. This ensures that the checkpoint stays in sync with the index checkpoint even if the checkpoint advance happened due to the documents deleted from the index.

## How to test this PR locally

1. Run the stack with the fleet server and the agent.
2. Place some actions, like for example changing the agent logging level, or osquery live query.
3. Delete the action document.
4. Check the fleet server logs. The fleet server should not go into the tight loop after the deletion.

## Design Checklist


## Checklist



## Related issues

- Closes https://github.com/elastic/fleet-server/issues/2205
